upf/upf_forward.c

```diff
@@ -211,10 +212,45 @@ upf_forward (vlib_main_t * vm, vlib_node_runtime_t * node,
 		  fp_entry =
 		    pool_elt_at_index (gtm->upf_forwarding_policies,
 				       far->forward.fp_pool_index);
-		  vnet_buffer (b)->ip.adj_index[VLIB_TX] =
-		    fp_entry->dpo.dpoi_index;
+
+		  vec_foreach (rpath, fp_entry->rpaths)
+		  {
+		    if (rpath->frp_proto ==
+			(is_ip4 ? DPO_PROTO_IP4 : DPO_PROTO_IP6))
+		      break;
+		  }
+
+		  if (!rpath)
```

Comment:
that test is not going to work. rpath will not be `NULL` if the vec_foreach reached the end of vector.

This might:
```suggestion
		  if (var < vec_end (rpath))
```

Reply:
added a check for `rpath < vec_end (rpaths)`